docs/blueprints/storage-node-downtime-disqualification.md

```diff
@@ -0,0 +1,107 @@
+# Storage Node Downtime Disqualification
+
+## Abstract
+
+This document describes storage node downtime suspension and disqualification.
+
+## Background
+
+The previous implementation of downtime disqualification led to the unfair disqualification of storage nodes. This prompted a halt to downtime disqualification and a new design for [Downtime Tracking](storage-node-downtime-tracking.md).
+
+While the previously mentioned blueprint presents a means of tracking downtime, it leaves out the design of how to use this information for disqualification. That is the purpose of this blueprint.
+
+## Design
+
+### Suspension
+
+Storage node downtime can have a range of causes. For those storage node operators who may have fallen victim to a temporary issue, we want to give them a chance to diagnose and fix it before disqualifying them for good. For this reason, we are introducing downtime suspension as a component of disqualification.
+
+For downtime suspension and disqualification we need a few new configurable values:
+
+- Tracking Period: The length of time into the past for which we measure downtime.
+- Allowed Downtime: The amount of downtime within the tracking period we will allow before suspension.
+- Grace Period: The length of time to give suspended SNOs to diagnose and fix issues causing downtime. Afterwards, they will have one tracking period to demonstrate an acceptable amount of downtime before disqualification.
+
+When the [downtime tracking system](storage-node-downtime-tracking.md) adds an entry to the nodes_offline_time table, if the node is not already suspended, we check the total amount of downtime for the tracking period, e.g. if the tracking period is 30 days, we sum downtime for the last 30 days. If the total is greater than the allowed downtime, the node is suspended for a length of time equal to the grace period + one tracking period.
```

Comment:
Only when the downtime tracking system is adding an entry? As far as I know we have 2 jobs. One job will identify offline nodes and add them to the table. Another job will ping these nodes and update the downtime of that entry. In case of an successful ping or storage node checkin we should evaluate the total downtime and put the node into suspension mode. We have to check it at the end of the ping process and not at the beginning.

Reply:
The first job will not add them to the table. It only calls overlay.UpdateUptime, which will set either last_contact_success or failure. The second job will ping nodes where last_contact_failure > last_contact_success. If it fails the ping, downtime is added to the table. That is when we check the total